### PR TITLE
refactor(semantic): do not resolve IdentifierReference used in TSTypeName

### PR DESCRIPTION
**Pull request by Dunqing**
TEST PR

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#3919** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/3919?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_semantic/src/builder.rs [hunk 1]
```diff
@@ -1650,7 +1650,7 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
 
     fn visit_ts_type_parameter(&mut self, ty: &TSTypeParameter<'a>) {
         let kind = AstKind::TSTypeParameter(self.alloc(ty));
-        self.enter_scope(ScopeFlags::empty());
+        // self.enter_scope(ScopeFlags::empty());
```
Comment (Dunqing) [inline crates/oxc_semantic/src/builder.rs:1653]: I think it was a workaround done before to get most of the tests passed.

**Comment by Dunqing** [comment]
I didn't find tests in TypeScript specifically for testing scopes

**Comment by Boshen** [comment]
https://github.com/oxc-project/oxc/blob/27665945f112b193f849fd611434436a83e0546a/crates/oxc_ast/src/visit/visit.rs#L1343-L1349

Try get rid of this as well

**Comment by Boshen** [comment]
We are studying how tsc does this. Closing for house keeping.

**Pull request closed without merging**